### ISSUE TITLE
run_eagerly argument bug in tf.Keras.Model compile method

### ISSUE THREAD

**Issue by Lillypucien** [tensorflow/tensorflow#37895]
**System information** 
- Have I written custom code (as opposed to using a stock
example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g.,
Linux Ubuntu 16.04): Windows 10, x64
- TensorFlow installed from (source or
binary): Binary (pip)
- TensorFlow version (use command below): TF 2.2.0 (2.2.0.dev20200324)
- Python version: 3.7
- CUDA/cuDNN version: 10.1 / 7.6
- GPU model and memory: Bug appears on several computers with different GPU

**Describe the current behavior**

The run_eagerly argument of the compile method of tf.Keras models does not actually set the run_eagerly attribute of the model. This causes the model to not be run eagerly even if run_eagerly=True was given as argument to its compile method.

This seems to be caused by a recent modification of the arguments of the compile method, as detailed bellow.

**Describe the expected behavior**

Calling compile on a Keras model with the argument run_eagerly=True should set the run_eagerly attribute of the model to True, which causes the model to be run eagerly when calling methods such as fit on it.

**Standalone code to reproduce the issue** 

```
import numpy as np
import tensorflow as tf

class CustomLayer(tf.keras.layers.Layer):
	def __init__(self):
		super().__init__()
		
	def call(self, inputs):
		tf.print('Running eagerly: ', tf.executing_eagerly())
		return inputs

if __name__ == "__main__" :
	data = np.random.random((16, 3)).astype(np.float32)

	inputs = tf.keras.Input(shape=(3,))
	outputs = tf.keras.layers.Dense(3)(inputs)
	outputs = CustomLayer()(outputs)
	model = tf.keras.Model(inputs=inputs, outputs=outputs)
	
	model.compile(loss='mse', run_eagerly=True) # does not set model.run_eagerly to True, and model does not run eagerly
	# model.run_eagerly = True # sets model.run_eagerly to True, and model runs eagerly

	print('run_eagerly: ', model.run_eagerly)

	model.fit(x=data, y=data)
```

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

Logs corresponding to the standalone code: [log.txt](https://github.com/tensorflow/tensorflow/files/4380372/log.txt)

**Supplementary info regarding the issue**

compile's signature is now
```
def compile(self,
              optimizer='rmsprop',
              loss=None,
              metrics=None,
              loss_weights=None,
              sample_weight_mode=None,
              weighted_metrics=None,
              run_eagerly=None,
              **kwargs):
```
when it used to be (notice that run_eagerly=None has been added to its arguments)
```
def compile(self,
              optimizer='rmsprop',
              loss=None,
              metrics=None,
              loss_weights=None,
              sample_weight_mode=None,
              weighted_metrics=None,
              **kwargs):
```

However, it still sets _run_eagerly as follows `self._run_eagerly = kwargs.pop('run_eagerly', None)` which can no longer work as run_eagerly is no longer in kwargs.

**Comment by goldiegadde** [comment]
the issue is not seen with 2.2.0-rc1, @Lillypucien can you try that version please

**Comment by jvishnuvardhan** [comment]
I can reproduce the issue with `tf-nightly`. [Here](https://colab.research.google.com/gist/jvishnuvardhan/ae6240266965ecb4803c920e67b64347/untitled40.ipynb) is the colab gist for y/our reference. Thanks

**Comment by Lillypucien** [comment]
The issue was indeed not yet in 2.2.0-rc1, but is still present in the latest tf-nightly as shown by @jvishnuvardhan

**Comment by Lillypucien** [comment]
@ymodak TF 2.0 is not the correct tag for this issue, the bug is present in the last tf 2.2 development version

**Comment by goldiegadde** [comment]
fixed the labels, this issue is not seen with 2.2.0-rc2 as well, but seen with latest nightly.

**Comment by goldiegadde** [comment]
This should have been fixed in this commit [ee5a324](https://github.com/tensorflow/tensorflow/commit/6f9a5289f9a59164f17a70029df9e9f39ee5a324).

**Comment by jvishnuvardhan** [comment]
@Lillypucien This was resolved in recent `tf-nightly`. [Here](https://colab.research.google.com/gist/jvishnuvardhan/19d45437d71b437c060057f8bb54b250/untitled40.ipynb) is the gist for your reference.

Can you please verify once and close the issue if this was resolved for you. Thanks!
